src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -814,6 +891,155 @@ public MergedBatchPutCommand ToCommand(DocumentsOperationContext context, Docume
             }
         }
 
+        internal class MergedBatchFixDocumentMetadataCommand : TransactionOperationsMerger.MergedTransactionCommand, IDisposable
+        {
+            private readonly Logger _log;
+            public HashSet<string> Ids = new HashSet<string>();
+            private readonly DocumentDatabase _database;
+            private readonly DocumentsOperationContext _context;
+            public DocumentsOperationContext Context => _context;
+            private bool _isDisposed;
+            public bool IsDisposed => _isDisposed;
+
+            public MergedBatchFixDocumentMetadataCommand(DocumentDatabase database, Logger log)
+            {
+                _database = database;
+                _log = log;
+                _database.DocumentsStorage.ContextPool.AllocateOperationContext(out _context);
+            }
+
+            protected override int ExecuteCmd(DocumentsOperationContext context)
+            {
+                if (_log.IsInfoEnabled)
+                    _log.Info($"Trying to update {Ids.Count:#,#0} documents metadata if necessary");
+
+                var count = 0;
+                foreach (var id in Ids)
+                {
+                    using (DocumentIdWorker.GetSliceFromId(context, id, out Slice lowerId))
+                    {
+                        if (_database.DocumentsStorage.GetTableValueReaderForDocument(context, lowerId, throwOnConflict: true, out var tvr) == false)
```

Comment:
This looks like `DocumentsStorage.Get`. We should not expose the method externally? Also probably you do not want to throw on conflict?